src/variables.cpp

```diff
@@ -9,47 +11,76 @@ VariablesAndValues *VariablesAndValues::getInstance()
 {
 	if (self == nullptr) {
 		self = new VariablesAndValues();
+
+		self->dataStorage = obs_data_create();
+		self->dataStoragePath = obs_module_config_path("variables.json");
+
+		auto *existingData = obs_data_create_from_json_file_safe(self->dataStoragePath, "bak");
+		obs_data_apply(self->dataStorage, existingData);
 	}
 	return self;
 }
 
+void VariablesAndValues::updateVariables(std::set<QString> updatedList) {
+	for(auto &variable : variables) {
+		if(!updatedList.count(variable)) {
+			obs_data_erase(dataStorage, variable.toUtf8());
+		}
+	}
+	
+	variables.clear();
+	variables = updatedList;
+}
+
 void VariablesAndValues::putVariable(const QString &variable)
 {
-	blog(LOG_INFO, "VariablesAndValues::putVariable %s",
-	     variable.toStdString().c_str());
-	if (variablesAndValues.find(variable) == variablesAndValues.end()) {
-		variablesAndValues[variable] = QString("");
+	blog(LOG_DEBUG, "VariablesAndValues::putVariable: Parsing found variable %s",
+	     variable.toStdString());
+
+	if(!variables.count(variable)) {
 		variables.insert(variable);
 	}
 }
 
 bool VariablesAndValues::contains(const QString &variable)
 {
-	return variables.find(variable) != variables.end();
+	return variables.count(variable) != 0;
 }
 
 void VariablesAndValues::putValue(const QString &variable, const QString &value)
 {
-	blog(LOG_INFO, "VariablesAndValues::putValue %s %s",
-	     variable.toStdString().c_str(), value.toStdString().c_str());
-	variablesAndValues[variable] = value;
-	variables.insert(variable);
+	blog(LOG_INFO, "VariablesAndValues::putValue: Variable %s, Value %s",
+	     variable.toStdString(), value.toStdString());
+
+	putVariable(variable);
+	obs_data_set_string(
+ 					dataStorage, variable.toUtf8(),
```

Comment:
has the clang formatter been run?

Reply:
Definitely clang doing whatever it wants.  I removed the extra spacing in my edits.